crates/core/src/contracts/erc20/ethereum_erc20.rs

```diff
@@ -49,4 +75,58 @@ impl<'a, P: Provider + Send + Sync> EthereumErc20<'a, P> {
             actual: balance.len(),
         })?)
     }
+
+    pub async fn name(self, block_id: BlockId) -> Result<String, EthApiError<P::Error>> {
+        // Prepare the calldata for the bytecode function call
+        let calldata = IERC20Calls::Symbol(SymbolCall).encode();
+        let calldata = calldata.into_iter().map(FieldElement::from).collect();
+
+        let block_id = EthBlockId::new(block_id);
+        let block_id: StarknetBlockId = block_id.try_into()?;
+
+        let result = self.kakarot_contract.eth_call(&self.address, calldata, &block_id).await?;
+        let name: Vec<u8> = result.0.into();
+
+        Ok(String::from_utf8(name.clone()).map_err(|_| DataDecodingError::InvalidReturnArrayLength {
+            entrypoint: "name".into(),
+            expected: 32,
```

Comment:
isn't name a string in solidity? it could be of arbitrary length, right?
i tried locally on remix and you can put very long names

Reply:
Oh yeah I had those as placeholders for the time being to get things to compile; looking at it again should I just use `SignatureDecodingError` or `TransactionDecodingError` instead?

Reply:
wdyt about adding another variant, like `DataDecodingError::InvalidBytesString` ?